test/uk/gov/hmrc/tradergoodsprofiles/connectors/RouteConnectorSpec.scala

```diff
@@ -151,6 +160,47 @@ class RouteConnectorSpec extends PlaySpec with ScalaFutures with EitherValues wi
       }
     }
   }
+
+  "post" should {
+
+    "return 201 when the record is successfully created" in {
+      val createRecordRequest = createRouterCreateRecordRequest()
+
+      when(httpClient.post(any)(any)).thenReturn(requestBuilder)
+      when(requestBuilder.setHeader(any)).thenReturn(requestBuilder)
+      when(requestBuilder.withBody(any[Object])(any, any, any)).thenReturn(requestBuilder)
+      when(requestBuilder.execute[HttpResponse](any, any)).thenReturn(Future.successful(HttpResponse(201, "message")))
+
+      val result = await(sut.post(createRecordRequest))
+
+      result.status mustBe CREATED
+    }
+
+    "send a request with the right url and body" in {
+      val createRecordRequest = createRouterCreateRecordRequest()
+
+      when(httpClient.post(any)(any)).thenReturn(requestBuilder)
+      when(requestBuilder.setHeader(any)).thenReturn(requestBuilder)
+      when(requestBuilder.withBody(any[Object])(any, any, any)).thenReturn(requestBuilder)
+      when(requestBuilder.execute[HttpResponse](any, any)).thenReturn(Future.successful(HttpResponse(201, "message")))
+
+      await(sut.post(createRecordRequest)(hc))
+
+      val expectedUrl = UrlPath.parse("http://localhost:23123/trader-goods-profiles-router/records")
+      verify(httpClient).post(eqTo(url"$expectedUrl"))(any)
+      verify(requestBuilder).setHeader(HeaderNames.CONTENT_TYPE -> MimeTypes.JSON)
+      verify(requestBuilder).setHeader("X-Client-ID"            -> "clientId")
+      verify(requestBuilder).withBody(eqTo(Json.toJson(createRecordRequest)))(any, any, any)
+      verify(requestBuilder).execute(any, any)
+
+      withClue("process the response within a timer") {
+        verify(metricsRegistry).timer(eqTo("tgp.createrecord.connector-timer"))
+        verify(metricsRegistry.timer(eqTo("emcs.submission.connector-timer"))).time()
```

Comment:
this is not TGP, should not be here. This may be my doing (copy and paste). Can this be removed please? I think there are several occurrences of this